server.js

```diff
@@ -14,15 +13,127 @@ mongoose.Promise = Promise
+    trim: true, 
+    minlength: 5, 
+    maxlength: 140
+    // validate: {
+    //   validator: (value) => {
+    //     return /^[^0-9]+$/.test(value);
+    //   }
+    // },
+  },
+  hearts: {
+    type: Number, 
+    default: 0
+  },
+  createdAt: {
+    type: Date, 
+    default: Date.now
+  }
+})
+
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
+// ROUTES
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hello thought project')
 })
 
+// GET REQUEST
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+  res.json(thoughts)
+})
+
+// POST REQUESTS
+// By doing await new Thought({ message }).save()instead of saving the entire body it's
+// only saving the message sent to the client other info will not be sent
+// adds a layer of security
+
+app.post('/thoughts', async (req, res) => {
+  try {
+    const { message } = req.body
+    const newThought = await new Thought({ message }).save()
+    res.json(newThought)
+  } catch (error) {
+    res.status(400).json({ sucess: false, error })
+  }
+})
+
+// increasing amount of hearts (likes)
+app.post('/thoughts/:id/likes', async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedThought = await Thought.findOneAndUpdate(
+      {
+        _id: id
+      },
+      { 
+        $inc: {
+          hearts: 1
+        }
+      },
+      {
+        new: true
+      }
+    );
+    if (updatedThought) {
+      res.json(updatedThought);
+    } else {
+      res.status(404).json({ message: 'Not found' })
+    }
+  } catch (error) {
+    res.status(400).json({ message: 'Invalid request', error });
+  }
+});
+
+// delete end point (WANTED TO CONTINUE WORKING ON THIS, THIS IS WHY IT IS COMMENTED OUT)
+// app.delete('/thoughts/:id', async (req, res) => {
+//   const { id } = req.params
+
+//   // find and delete 
+//   try {
+//     const deletedThought = await Thought.findOneAndDelete({ _id: id })
+//     if (deletedThought) {
+//       res.json(deletedThought)
+//     } else {
+//       res.status(404).json({ message: "not found" })
+//     }
+//   } catch {
+//     res.status(400).json({ message: "invalid request", error: "error" })
+//   }
+// })
+
+// update entity "Patch", to replace, "Put" we can us findByIdAndReplace, and whole object ubdate
+// app.patch('thoughts/:id', async (req, res) => {
+//   const { id } = req.params
+
+//   try {
+// const updatedThought = await Thought.findByIdAndUpdate
+// (id, { name: 'req.body.name' }, { new: true })
+//     if (updatedThought) {
+//       res.json(updatedThought)
+//     } else {
+//       res.status(404).json({ message: "not found" })
+//     }
+//   } catch {
+//     res.status(400).json({ message: "invalid request", error: "error" })
+//   }
+// })
+
```

Comment:
remember to always restrict access to all endpoint which can manipulate your data stored in DB.